Update the Readme File (#1)
Kicchu02/Draft-Craft: README.md

> ## Draft-Craft

Reviewer: ```suggestion
## Draft-Craft
```
# Draft-Craft

> ## Draft-Craft

Reviewer: ```suggestion
# Draft-Craft
```
Heading should always be H1.

This project is a notes app built using React and Electron. The app features a simple, user-friendly interface designed for productivity and note management.

> ### Setting Up the Project

Reviewer: ```suggestion
## Setting Up the Project
```
Why is this H3 while the headings that come under it are H2?

> ## Prerequisites

Reviewer: ```suggestion
### Prerequisites
```
This heading comes under `Setting Up the Project`. The heading style should be one below of it.

Before starting, ensure that you have the following tools installed:

1. **Node.js** (v20)

   - [Official Download Page](https://nodejs.org/en/download/)
   - Installation Instructions: [How to Install Node.js](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)

2. **Git**
   - [Official Download Page](https://git-scm.com/downloads)
   - Installation Instructions: [How to Install Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)

## Installation

> - Install the necessary dependencies:

Reviewer: ```suggestion
- Bootstrap the `npm` project:
```

```bash
npm install
```

- Run application

```bash
npm start
```

> The React application is located in the `src/renderer` directory. This directory contains all the necessary components, utilities, styles, and other resources for the application.

Reviewer: ``` suggestion
The React application is located in the `src/renderer` directory, which contains all the UI-related files.
```

> The source code for the React app is located in the `src/renderer` directory. This directory contains all the necessary components, utilities, styles, and other resources for the application.

Reviewer: I don't think you understood what is going on here. I'd like you to please understand the structure of an electron app. An app built in the electron framework is packaged with an isolated version of the Chromium engine. Basically, every app is a Chrome browser. This is done to make the UI using HTML, CSS and JS. The entry point of the application lives in the `src/main` directory. But only the UI is living in the `src/renderer` directory. In our case, the UI is built using React. When someone wants to work on this project, the main focus would be the UI. Hence, noting that the UI lives in this folder will be helpful. But suppose we wanna perform other operations like opening a file, displaying a folder etc, anything that is related to the OS itself. In that case, that logic must be written in a different folder. 

> This directory contains all the necessary components, utilities, styles, and other resources for the application.

So in that light, this sentence is wrong. Fix it.

> The source code for the React app is located in the `src/renderer` directory. This directory contains all the necessary components, utilities, styles, and other resources for the application.

Reviewer: ```suggestion
The React application lives in the `src/renderer` directory. This directory contains all the necessary components, utilities, styles, and other resources for the application.
```

> # Notes App

Reviewer: This must be `Draft Craft`. Make this change everywhere.

> This project is a notes app built using React and Electron. It allows users to create, organize, and search through their notes efficiently. The app features a simple, user-friendly interface designed for productivity and note management.

Reviewer: > It allows users to create, organize, and search through their notes efficiently.

Remove this sentence as we are not promising anything as of now.

Author: I have provided a brief description of the app in the recent commit

## Features

- **Markdown-based Note-Taking**: Create, edit, and save notes using markdown.
- **File Organization**: Organize notes into folders and subfolders.

> - **Search Functionality**: Quickly search through notes.

Reviewer: Feels like too much promise. I'm not sure we will take it up. Can u make this more generic and open. When we add specific features we can update the description to be more verbose.

Author: I have updated the readme file, pls check once.

## Prerequisites

Before starting, ensure that you have the following tools installed:

> 1. **Node.js** (v14 or later)

Reviewer: This project uses the latest LTS version, that is Node 20.

Author: fixed this one

> 2. **npm** or **yarn**

Reviewer: This is only an npm project. We are not supporting yarn.

Author: I have modified this part

> 3. **Git**

Reviewer: Please provide links to the official dowloads, or how to install page for each pre-requisite.

Author: I have provided the links to install the prerequisites in the recent commit

## Clone the Repository

> First, clone the repository and install the necessary dependencies:

Reviewer: This change was not necessary.

Author: noted

```bash
git clone --depth 1 --branch main https://github.com/electron-react-boilerplate/electron-react-boilerplate.git your-project-name
cd your-project-name
npm install

> ```

Reviewer: This needs to be updated. The repo is pointing to his GitHub. It should be ours.

Author: fixed this one

## Starting Development

> Use this command to start the development server:

Reviewer: This also need not be changed. This is desktop app. There is no server involved.

Author: noted

```bash
npm start
```

This will launch the app in development mode, where you can see changes live as you develop.

## Packaging for Production

To package the app for production on your local platform, run the following command:

```bash
npm run package

> ```

Reviewer: Does this really work. Have u tested it? I was thinking to first focus on development. Packaging and distribution can be looked upon later.

## Draft-Craft

This project is a notes app built using React and Electron. The app features a simple, user-friendly interface designed for productivity and note management.

## Features

- **Markdown-based Note-Taking**: Create, edit, and save notes using markdown.

> - **File Organization**: Organize notes into folders and subfolders.

Reviewer: Remove this section. We discussed this right?

Author: noted, pls check

## Prerequisites

Before starting, ensure that you have the following tools installed:

1. **Node.js** (v20)

   - [Official Download Page](https://nodejs.org/en/download/)
   - Installation Instructions: [How to Install Node.js](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)

2. **Git**
   - [Official Download Page](https://git-scm.com/downloads)
   - Installation Instructions: [How to Install Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)

## Installation Steps

- First, clone the repository and install the necessary dependencies:

```bash
git clone https://github.com/Kicchu02/Draft-Craft.git

> cd Draft-Craft

Reviewer: Remove the clone part. We discussed this as well [here](https://github.com/Kicchu02/Draft-Craft/pull/1#issuecomment-2395024189)

npm install
```

- Run application

```bash
npm start
```

### Directory Structure

The source code for the React app is located in the `src/renderer` directory. This directory contains all the necessary components, utilities, styles, and other resources for the application.

Here’s a brief overview of the folder structure:

```bash
Draft-Craft/
├── src/                    # Source code of the application
│   ├── renderer/           # React app source code (located here)
│   │   ├── components/     # React components
│   │   ├── utils/          # Utility functions and helpers
│   │   ├── styles/         # CSS/SCSS files for styling
│   │   └── assets/         # Static assets like icons or images
│   ├── main/               # Electron main process files
│   └── ...
├── public/                 # Public static files
├── package.json            # App configuration and dependencies
├── webpack.config.js       # Webpack bundler configuration
└── README.md               # Project documentation (this file)

> ```

Reviewer: This is really not necessary. We will be constantly updating the project. As we write code, we cannot keep updating this and keep track of it. Please remove it.
